### PR TITLE
Allow crossbows to be used in practice recipes

### LINKED ISSUE #60498: Crossbow training with archery targets.
### Is your feature request related to a problem? Please describe.

We have practice recipes for bows, cool! That's really good for us. ...Let us do the same with crossbows.

### Solution you would like.

Just give similar practice recipes to crossbows for rifles training. Nothing much here.

### Describe alternatives you have considered.

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by MNG-cataclysm**
#### Summary
Content "Allow for crossbows to be used in practice recipes."

#### Purpose of change
Allow for crossbows to be used in practice recipes to train gun skills.

#### Describe the solution
Added two new practice recipes for crossbows. They work like archery practice, requiring the use of bolts in order to train the rifle skill. 

#### Describe alternatives you've considered
Giving crossbows a rifle quality to be used in the targeting game, like other guns.

#### Testing
None

#### Additional context
Closes #60498

### CODE REVIEW

**Review by Maleclypse — APPROVED** [review]

File: data/json/recipes/practice/ranged.json [hunk 1]
```diff
@@ -61,6 +61,24 @@
...
+    "category": "CC_PRACTICE",
+    "subcategory": "CSC_PRACTICE_COMBAT",
+    "name": "crossbows (beginner)",
+    "description": "Practice firing unfletched bolts at a spot on the wall.",
+    "skill_used": "rifle",
+    "time": "1 h",
+    "practice_data": { "min_difficulty": 0, "max_difficulty": 1, "skill_limit": 2 },
+    "autolearn": [ [ "rifle", 1 ] ],
+    "book_learn": [ [ "recipe_bows", 0 ], [ "manual_archery", 0 ], [ "book_archery", 0 ] ],
+    "tools": [ [ "bullet_crossbow", "crossbow", "compositecrossbow", "compcrossbow", "huge_crossbow" ] ],
```
Comment (Fris0uman) [inline data/json/recipes/practice/ranged.json:78]: Should put those into a requirement and use that instead I think, that way when more crossbows are added one can just go edit the requirement and all training recipes will update.

File: data/json/recipes/practice/ranged.json [hunk 2]
```diff
@@ -95,6 +113,27 @@
...
+    "subcategory": "CSC_PRACTICE_COMBAT",
+    "name": "crossbows (intermediate)",
+    "description": "Practice the best ways of firing fletched bolts into a target.  Requires a decent quality crossbow.",
+    "skill_used": "rifle",
+    "time": "30 m",
+    "practice_data": { "min_difficulty": 2, "max_difficulty": 4, "skill_limit": 5 },
+    "autolearn": [ [ "rifle", 3 ] ],
+    "book_learn": [ [ "recipe_bows", 2 ], [ "manual_archery", 2 ], [ "book_archery", 2 ] ],
+    "tools": [
+      [ "bullet_crossbow", "crossbow", "compositecrossbow", "compcrossbow", "huge_crossbow" ],
```
Comment (Fris0uman) [inline data/json/recipes/practice/ranged.json:131]: Make that a crossbow requirement

File: data/json/recipes/practice/ranged.json (hunk 1 above)
Comment (MNG-cataclysm) [reply data/json/recipes/practice/ranged.json]: I've changed it to a requirement entry.

File: data/json/recipes/practice/ranged.json (hunk 2 above)
Comment (MNG-cataclysm) [reply data/json/recipes/practice/ranged.json]: Same as before, I've changed it to a requirement entry.

**Pull request merged**